internal/errors/option.go

```diff
@@ -24,3 +24,21 @@ var (
 		return Errorf("invalid option. name: %s, val: %#v", name, val)
 	}
 )
+
+// ErrCriticalOption represent the critical option error
+type ErrCriticalOption struct {
+	Err error
+}
+
+func (e *ErrCriticalOption) Error() string {
+	return e.Err.Error()
+}
```

Comment:
sorry. If I make a mistake.
I think the following is better.
what do you think?
```
type ErrCriticalOption struct {
    err error
}

func NewErrCriticalOption(err error) error {
    return &errCriticalOption {
       err: errr,
    }
}
```

I would like to avoid writing the following at the point of error generation.
```
func WithHost(host ...) Option {
    return func() error {
        if len(host) == 0 {
                return &ErrCriticalOption {
                         Err: err
                }
        }
    }
}
```

I think the following is simple.
```
```
func WithHost(host ...) Option {
    return func() error {
        if len(host) == 0 {
                return &NewErrCriticalOption(err)
        }
    }
}
```
```